portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/oauth2/OAuth2.testcase

```diff
@@ -22,242 +22,127 @@ definition {
-				websiteURL = "http://www.liferay.com");
-		}
-
-		task ("Input new values") {
-			var generatedClientId = selenium.getElementValue("OAuth2#ADMIN_INPUT_FORM_CLIENT_ID");
-			var generatedClientSecret = selenium.getElementValue("OAuth2#ADMIN_INPUT_FORM_CLIENT_SECRET");
-
-			OAuth2.inputApplicationValues(
-				description = "Registering a new application",
-				privacyPolicyURL = "http://www.liferay.com");
-
-			Navigator.gotoBack();
-		}
+	@description = "This is a use case for LPS-129398. Remember device with multiple auth without session."
+	@priority = "4"
+	test ApplicationCanBeAuthorizedWithoutUserSession {
+		property test.name.skip.portal.instance = "OAuth2#ApplicationCanBeAuthorizedWithoutUserSession";
 
-		task ("Assert correct application name, client Id and description") {
-			OAuth2.viewApplicationInTable(
-				applicationName = "OAuth Application",
-				clientId = "${generatedClientId}",
-				description = "Registering a new application");
-		}
+		task ("Add and edit 2 external applications") {
 
-		task ("Check OAuth application informations") {
-			OAuth2.viewApplication(
-				applicationName = "OAuth Application",
-				callbackURI = "http://www.liferay.com",
-				checkboxCheckList = "Authorization Code,Client Credentials,Refresh Token,Resource Owner Password Credentials",
-				clientId = "${generatedClientId}",
-				clientProfile = "Web Application",
-				clientSecret = "${generatedClientSecret}",
-				description = "Registering a new application",
-				privacyPolicyURL = "http://www.liferay.com",
-				websiteURL = "http://www.liferay.com");
-		}
+			// Test Setup
 
-		task ("Edit all application values") {
-			OAuth2.editApplication(
-				applicationName = "OAuth Application",
-				callbackURI = "http://www.liferay.com/callback",
-				checkboxCheckList = "Token Introspection",
-				checkboxUncheckList = "Refresh Token,Resource Owner Password Credentials",
-				clientId = "abcd1234",
-				clientProfile = "Native Application",
-				clientSecret = "abcd1234",
-				description = "Editing a new application",
-				newApplicationName = "OAuth Application Edited",
-				privacyPolicyURL = "http://www.google.com",
-				websiteURL = "http://www.google.com");
+			for (var count : list "1,2") {
+				OAuth2.addAndEditExternalApplication(
+					applicationName = "My application${count}",
+					callbackURI = '''http://martamedio.com/oauth2-tester/?url=${portalURL}&client_id=default${count}''',
+					checkboxCheckList = "Remember Device",
+					clientId = "default${count}",
+					clientProfile = "Web Application",
+					clientSecret = "defaultsecret${count}",
+					websiteURL = "http://www.liferay.com");
+			}
 		}
 
-		task ("View recent changes applied") {
-			Navigator.gotoBack();
-
-			OAuth2.viewApplicationInTable(
-				applicationName = "OAuth Application Edited",
-				clientId = "abcd1234",
-				description = "Editing a new application");
+		task ("Logout, authorize application1 and launch request") {
```

Comment:
Wording suggestion: "Logout then authorize application1 and launch request" - it sounds like a flow, rather than a list of actions :).